doc/dev/communication.rst

```diff
@@ -0,0 +1,251 @@
+                   auto sources = std::equal_range(sp, se, src, spike_pred());
+                   // Consequently, the range returned is the range of equal spike sources,
+                   // we pick out ours and add all of them to the appropriate queue.
+                   for (auto s: util::make_range(sources)) queues[cix].push_back(cn->make_event(s));
+                   // now, move to next
+                   sp = sources.first;
+                   ++cn;
+               }
+           }
+           else {
+               while (cn != ce && sp != se) {
+                   auto targets = std::equal_range(cn, ce, sp->source);
+                   for (auto c: util::make_range(targets)) queues[c.index_on_domain()].push_back(c.make_event(*sp));
+                   cn = targets.first;
+                   ++sp;
+               }
+           }
+       }
+   }
+
+After ``make_event_queues`` there is one queue per cell and each queue
+is filled with a time ordered list of events for that cell. We now need
+to understand the actual connection table stored in
+
+.. code-block:: c++
+
+   struct connection {
+       spike_event make_event(const spike& s) {
+           return { destination_, s.time + delay_, weight_};
+       }
+
+       cell_member_type source;
+       cell_lid_type destination;
+       float weight;
+       float delay;
+       cell_size_type index_on_domain;
+   };
+
+   struct communicator {
+       // [...]
+       cell_size_type num_domains_;
+       std::vector<connection> connections_;
+       std::vector<cell_size_type> connection_part_;
+       // [...]
+   };
+
+The ``connections`` vector is a list of connections partitioned by the
+domain (as in domain decomposition) of their source's ``gid``, while
+``connection_part`` stores the partioning indices.
+
+Building the Connection Table
+=============================
+
+The table of connections on the local rank is built during the
+connection of the ``communicator`` object
+
+.. code-block:: c++
+
+   communicator::communicator(const recipe& rec,
+                              const domain_decomposition& dom_dec,
+                              const label_resolution_map& source_resolution_map,
+                              const label_resolution_map& target_resolution_map,
+                              execution_context& ctx);
+
+After that process,
+
+.. code-block:: c++
+
+   struct communicator {
+       // ...
+       std::vector<connection> connections_;
+       std::vector<cell_size_type> connection_part_;
+   };
+
+will contain all connections in ``connections_`` partitioned by the
+domain of the source's ``gid`` in ``dom_dec``. Beginnings of the
+respective partitions are pointed to by the indices in
+``connection_part_``.
+
+The algorithm for building is slightly obscured by caching and the use
+of labels and resolving them via ``target_/source_resolution_map`` to
+local ids on the respective source and target cells.
```

Comment:
These `label_resolution_map` were a bit confusing - are they documented somewhere?

Reply:
Unfortunately not, they are an internal thing added during the switch 
```cxx
// Old
d = A.decor{};
d.place(...);
d.place(...);
do_something_to_item_two((0,1)); 

// New
d = A.decor{};
d.place(..., "item1");
d.place(..., "item2");
do_something_to_item_two("item2");
```
These maps translate textual labels, eg `"item"` into cell local addresses `(cell, item)`.

I can add the above explanation to a note/sidebar thing though.